File: app/src/main/java/social/entourage/android/encounter/CreateEncounterModule.java
package social.entourage.android.encounter;

import dagger.Module;
import dagger.Provides;

/**
 * Module handling all ui related dependencies
 */
@Module
public final class CreateEncounterModule {
    private final CreateEncounterActivity activity;

    public CreateEncounterModule(final CreateEncounterActivity activity) {
        this.activity = activity;
    }

    @Provides
    public CreateEncounterActivity providesActivity() {
        return activity;
    }
}

File: app/src/main/java/social/entourage/android/tour/encounter/CreateEncounterModule.kt
package social.entourage.android.tour.encounter

import dagger.Module
import dagger.Provides

/**
 * Module related to CreateEncounterActivity
 * @see CreateEncounterActivity
 */
@Module
internal class CreateEncounterModule(private val activity: CreateEncounterActivity) {
    @Provides
    fun providesActivity(): CreateEncounterActivity {
        return activity
    }

}